### ISSUE TITLE
Spurious test in TestInfo: domain_test.go:161: start syncer failed

### ISSUE THREAD

**Issue by kennytm** [pingcap/tidb#14427]
Sometimes the test in `domain` spuriously fail with messages like

```
[2020-01-09T07:20:17.793Z] === RUN   TestInfo
[2020-01-09T07:20:17.793Z] {"level":"warn","ts":"2020-01-09T15:20:05.732+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-aeb63d1c-da6d-4aec-aae6-2cffef8c315f/localhost:74629653422275528650","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
[2020-01-09T07:20:17.793Z] --- FAIL: TestInfo (1.19s)
[2020-01-09T07:20:17.793Z]     domain_test.go:161: start syncer failed
[2020-01-09T07:20:17.793Z] FAIL
[2020-01-09T07:20:17.793Z] coverage: 81.0% of statements
[2020-01-09T07:20:17.793Z] FAIL	github.com/pingcap/tidb/domain	1.599s
```

and

```
[2020-01-09T05:16:19.698Z] === RUN   TestInfo
[2020-01-09T05:16:19.698Z] {"level":"warn","ts":"2020-01-09T13:15:58.866+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-a7493aef-aa17-40b5-bf0a-b2ab72c1ccd5/localhost:68172685320723204620","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
[2020-01-09T05:16:19.698Z] {"level":"warn","ts":"2020-01-09T13:15:59.330+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-21d504f9-3c1b-49f5-941f-b412beaa548c/localhost:68172685320723204620","attempt":0,"error":"rpc error: code = NotFound desc = etcdserver: requested lease not found"}
[2020-01-09T05:16:19.698Z] --- FAIL: TestInfo (7.21s)
[2020-01-09T05:16:19.698Z]     domain_test.go:161: start syncer failed
[2020-01-09T05:16:19.698Z] FAIL
[2020-01-09T05:16:19.698Z] FAIL	github.com/pingcap/tidb/domain	10.734s
```

Maybe a duplicate of #12685.

**Comment by SunRunAway** [comment]
Another appearance in the newly master branch

https://github.com/pingcap/tidb/pull/14682#issuecomment-585614634
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/23990/pipeline

**Comment by SunRunAway** [comment]
It still happens
Seen at https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/27187/pipeline in https://github.com/pingcap/tidb/pull/14887

**Comment by Deardrops** [comment]
I try to fix this Issue in #15305.
